pkg/serverless/trace/propagation/carriers.go

```diff
@@ -253,6 +253,16 @@ func headersCarrier(hdrs map[string]string) (tracer.TextMapReader, error) {
 	return tracer.TextMapCarrier(hdrs), nil
 }
 
+// headersOrMultiheadersCarrier returns the tracer.TextMapReader used to extract
+// trace context from a Headers field of form map[string]string or MultiValueHeaders
+// field of form map[string][]string.
+func headersOrMultiheadersCarrier(hdrs map[string]string, multiHdrs map[string][]string) (tracer.TextMapReader, error) {
+	if len(hdrs) > 0 {
+		return headersCarrier(hdrs)
+	}
+	return tracer.HTTPHeadersCarrier(multiHdrs), nil
```

Comment:
Just a question -- why do we use the `TextMapCarrier` and `HTTPHeadersCarrier` from the Go tracer? Seems odd that the Go tracer is a dependency of the agent

Reply:
We rely on the tracer for trace context extraction. We create one of these "carrier" objects, then pass it to the `Propagator.Extract` method defined by the tracer. By relying on the tracer to do this work, it means we don't need to implement the complex logic to do it ourselves.